include/head/core/presence_sensing_driver.hpp

```diff
@@ -26,8 +96,42 @@ class PresenceSensingDriver {
         return voltage_read;
     }
 
+    auto static get_tool(adc::MillivoltsReadings reading) -> AttachedTool {
```

Comment:
this method doesn't modify any state and can be `const` i believe. Although I will also say, this method really just appears to be a constructor of the `AttachedTool` class -  maybe that's just what it should be.